### ISSUE TITLE
Welcome Guide: modal should be labelled 

### ISSUE THREAD

**Issue by enriquesanchez** [WordPress/gutenberg#19078]
**Describe the bug**
From https://github.com/WordPress/gutenberg/pull/18041#issuecomment-562661218

>The modal is not labelled. This is in part a problem with this modal, but should probably be corrected in the component: the Modal component allows the component to be misused.

>We should enforce a label for the modal element with `role=dialog` within the e2e tests, because all modals must be labelled. Additionally, the modal component should enforce `role="document"` on the content of all modal dialogs. This isn't strictly required on all modals, but the component can't know what type of content a modal will contain, this will give the best overall experience.
